parser/src/parser/tests/expressions.rs

```diff
@@ -5,6 +5,33 @@ use super::{
 
 // SECTIONS
 // ================================================================================================
+#[test]
+fn exponentiation() {
+    // the operation must be put into a source section, or parsing will fail
+    let source = "
+    transition_constraints:
+        enf clk'^2 + clk - 1 = 0";
```

Comment:
Same thoughts I left in the `Mul` PR about more rigorous testing of edge cases & different combinations of operations.